src/main/java/fi/aalto/cs/apluscourses/utils/observable/ObservableProperty.java

```diff
@@ -38,8 +49,12 @@
     callback.valueChanged(observer, get());
   }
 
+  public <O> void addValueObserver(@NotNull O observer, @NotNull SimpleCallback<O> callback) {
+    addValueObserver(observer, new CallbackWrapper<>(callback));
+  }
+
   public void declareDependentOn(@NotNull ObservableProperty<?> property) {
-    property.addValueObserver(this, (self, dummy) -> self.onValueChanged(self.get()));
+    property.addValueObserver(this, ObservableProperty<T>::onValueChangedInternal);
```

Comment:
I liked the lambda more :smile:

Reply:
Ah yes, this was because I though I'd need `onValueChangedInternal` for something else too but ended up removing that use of it so it only was referenced here. I agree, it is unnecessary for this single purpose, so I'll reintroduce that lambda.

Reply:
Although that lambda is a bit tricky because one have to be careful not to write e.g.
```java
(self, dummy) -> self.onValueChanged(this.get())
```
but we are careful, right?

Reply:
Good point :smile: I'm sure once it's there, no one will change it without being careful.... hopefully..